transport/server.cc

```diff
@@ -634,6 +655,12 @@ future<> cql_server::connection::process_request() {
 
             return make_ready_future<>();
           });
+        }).handle_exception([this, length = f.length] (const std::exception_ptr&) {
+            // Cancel shedding in case no more requests are going to do that on completion
+            if (_pending_requests_gate.get_count() == 0) {
+                _shed_incoming_requests = false;
+            }
+            return _read_buf.skip(length);
```

Comment:
This is just to handle the timeout, yes? It should be mentioned and maybe even use handle_exception_type.

The extra continuation is unfortunate. Although, if there is no contention, it doesn't generate a task.

I hesitate about recommending to coroutinize this, since process_request()'s inner continuation (after read_frame()) will not wait in the common case.  We can't even fold it into the per-connection code since that's in some generic server.

Reply:
Its main purpose is for the timeout, but I intentionally put a generic `handle_exception` here, since *any* error (including some future ones) happening before the deferred action above (called `leave`) could leave the mechanism in a permanent shedding state - since there would be nobody to stop the process.

But, I now see that skipping `length` bytes in this handler is too eager - these bytes might have already been read by `read_and_decompress_frame`... I'd need to make this handler more robust.

Reply:
I'll solve this one by attaching the additional continuation right after getting semaphore units with a timeout. Additional bonus is that only the shedding path will now include this continuation.